server/src/main/java/org/elasticsearch/index/mapper/ValueFetcher.java

```diff
@@ -46,6 +47,24 @@
      */
     List<Object> fetchValues(SourceLookup lookup) throws IOException;
 
+    /**
+    * Given access to a document's _source, return this field's values.
+    *
+    * In addition to pulling out the values, they will be parsed into a standard form.
+    * For example numeric field mappers make sure to parse the source value into a number
+    * of the right type.
+    *
+    * Note that for array values, the order in which values are returned is undefined and
+    * should not be relied on.
+    *
+    * @param lookup a lookup structure over the document's source.
+    * @param ignoredFields the fields in _ignored that have been ignored for this document because they were malformed
+    * @return a list a standardized field values.
+    */
+    default List<Object> fetchValues(SourceLookup lookup, Set<String> ignoredFields) throws IOException {
```

Comment:
Sure, I was trying to avoid changing this for all ValueFetchers, but trying in c342c64 now and the diff doesn't look too bad. Happy to go this way if you also don't mind the extra (mostly unused) argument in those places.